.github/workflows/trigger-release-candidate-build-and-deploy.yml

```diff
@@ -0,0 +1,39 @@
+name: Trigger release candidate build and deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main branch and staging branch
+        uses: actions/checkout@v2
+        with:
+          ref: main
+          fetch-depth: 0 # fetch all history to make merges work correctly
+      - name: Merge main branch into staging branch
```

Comment:
I thought the plan was deploy main to the staging environment and removing the staging branch?

Reply:
@Bento007 (cc: @atolopko-czi ) -

Here we want to do the simplest thing to demonstrate the pipeline end-to-end. It is not the most efficient, but deploying `main` to `staging` and reusing the existing `deploy-happy-stack.yml` to deploy to `stage` env seems to be the simplest thing to do.

The most efficient route is to build only once and deploy the builds to different environments (`staging`, `prod`, etc) but that is a bit complex in terms of implementation and testing.

There are some additional benefits to this simple approach too:

1. Easier to test. GHA, as far as I understand, is pretty onerous to test. The number of commits in this PR to demonstrate auto-merge is an example of the pain point. The simplest path to the solution also happens to be the shortest path - just merge `main` to `staging` - which means easier to demonstrate a test of correctness.
2. Concurrency introduces complexity (ex: many commits to `main` coming in at once) and to the extent we can keep things straightforward, will allow us to understand the complexities of concurrency and then we can determine a more efficient solution
3. Easier to reason about in the beginning (Uniformity) - The merge-commit that goes from `main` to `staging` is replicates our documented manual steps for a `stage` deploy thus making continuous deployment to staging easier to reason about.
4. Easy to rollback. If continuous deployment to `stage` is running into problems, it is as simple as disabling the GHA workflow that merges `main` to `staging` and we are back to our starting point.

Reply:
I agree w/most of this and the overall decision.

For 2, haven't we addressed the concurrency issue with `concurrency` config? If not, isn't the concurrency concern for `staging` branch merges going to be the same as it would be for `main`-based staging deploys?

For 1, most of the GHA-dev/debug cycle was for the main->staging auto-merge right? So possibly might have been shorted dev time if deployed to stagin from `main`? (I haven't studies all the commits!)

Reply:
@atolopko-czi 

For concurrency (2), we have a provisional solution with `concurrency` feature but we haven't tested it. My point is that it is hard to see right now what kinds of concurrency gotchas we might run into.

For the testing point made in (1), if I understand you correctly, you are proposing committing the change to `main` and then see if it actually auto-merges to `staging`? In that case it would take the same number of attempts (see the number of commits to test auto-merge between test branches) and moreover, pollute `main` with a bunch of these "attempting-fix" commits.

In the absence of unit tests, I think it is preferable to demonstrate tests in testing sandbox - in this case, using a couple of test branches

Reply:
> For the testing point made in (1), if I understand you correctly, you are proposing committing the change to main and then see if it actually auto-merges to staging?  

Not actually proposing anything :) We can take this offline, since I was just thinking aloud.